client/src/config/mod.rs

```diff
@@ -167,4 +169,25 @@ socket = "/var/run/aurae/aurae.sock"
             matches!(config.system.socket, AuraeSocket::Path(path) if Some("/var/run/aurae/aurae.sock") == path.to_str())
         )
     }
+
+    #[test]
+    fn can_parse_toml_config_socket_ipv6() {
+        const INPUT: &str = r#"
+[auth]
+ca_crt = "~/.aurae/pki/ca.crt"
+client_crt = "~/.aurae/pki/_signed.client.nova.crt"
+client_key = "~/.aurae/pki/client.nova.key"
+
+[system]
+socket = "[fe80::2]:8080%4"
```

Comment:
nit: we should also test not having the scope id as it is optional.

Reply:
In system_config.rs there is `can_parse_aurae_socket_ipv6` and `can_parse_aurae_socket_ipv6_with_scope_id` which should cover both scenarios. 

I was actually going to take this test out, but forgot :). I'm not sure it adds anything. over the path one, but if you see value in it I can add the not scope one here too.